### ISSUE TITLE
Allow to enter domain instead of the ip address of the server

### ISSUE THREAD

**Issue by mkhib** [amnezia-vpn/amnezia-client#892]
In case that the IP gets blocked in the country we can change the IP behind the domain instead of sharing new connections to the users.

**Comment by Tumist76** [comment]
Upvoting on this one. Have been rotating servers previously because of connection low speed and without hostname it's a pain.

**Comment by amzaf** [comment]
In my case, ipv4 address of my server is blocked but I can still access it through ipv6. Hiding the real server address behind Cloudflare is another good practice which necessitates using a domain name.

**Comment by sbh96** [comment]
you must add domain support if the user prefers can use it based on his region or make a subscription link (like your api to provide free subscriptions ) auto reval to update from server side every 15 mints to make update if any things changes in server  side

**Comment by Nethius** [comment]
If someone can test this branch https://github.com/amnezia-vpn/amnezia-client/pull/814, it will help us a lot. Domain name resolve should work for awg/wg protocols

**Comment by virtualos** [comment]
> If someone can test this branch #814, it will help us a lot. Domain name resolve should work for awg/wg protocols

I could download this dev build, Thank you, I am testing...

**Comment by virtualos** [comment]
**I tried AmneziaWG and Xray with and without Amnezia DNS service and I got "Error Code 1000" everytime.**

**Comment by virtualos** [comment]
> If someone can test this branch #814, it will help us a lot. Domain name resolve should work for awg/wg protocols

**I used Dev version which built in repo**

- Device -> Mobile Android 14
- Domain -> xxx.com
- Used Cloudflare for A record to connect xxx.com to server IP (Not proxified).
- Is Amnezia succeed to install Xray and AmneziaWG? -> YES
- Did VPN connect? -> NO, Error Code 1000

**Comment by virtualos** [comment]
> If someone can test this branch #814, it will help us a lot. Domain name resolve should work for awg/wg protocols

I tested dev build:
**added the ability to enter a domain name when configuring the server #3161**
https://github.com/amnezia-vpn/amnezia-client/actions/runs/10318395547/artifacts/1794715477

Server: Hetzner Data center 
OS: Ubuntu 22.04
Amnezia: Android
Domain: 94web.com and in Cloudflare domain linked to server IP by A record with No proxy!

Summery:
I successfully installed AmneziaWG but when I tried to connect I got 1000 error code.

Here is Amnezia Log report:

`Model: A063
Brand: Nothing
Product: Spacewar
Device: Spacewar
Codename: REL
Release: 14
SDK: 34
ABI: arm64-v8a, armeabi-v7a, armeabi

08-18 05:25:46.973 21590 21614 D [qtMainLoopThread] AmneziaQt: Call activity method: exportLogsFile

LOGCAT:
--------- beginning of main
08-18 05:23:20.340 20484 20484 E org.amnezia.vpn: Not starting debugger since process cannot load the jdwp agent.
08-18 05:23:20.673 20484 20507 E OpenGLRenderer: Unable to match the desired swap behavior.
08-18 05:23:33.508 20484 20507 E OpenGLRenderer: Unable to match the desired swap behavior.
08-18 05:23:40.281 20484 20507 E OpenGLRenderer: Unable to match the desired swap behavior.
08-18 05:25:12.142 21163 21163 E org.amnezia.vpn: Not starting debugger since process cannot load the jdwp agent.
08-18 05:25:12.370 21163 21185 E OpenGLRenderer: Unable to match the desired swap behavior.
08-18 05:25:16.150 21163 21185 E OpenGLRenderer: Unable to match the desired swap behavior.
08-18 05:25:18.299 21304 21304 E neziaAwgService: Not starting debugger since process cannot load the jdwp agent.
08-18 05:25:19.145 21304 21325 E AmneziaVpnService: Not a numeric address: 94web.com
08-18 05:25:19.151 21163 21163 E AmneziaActivity: From VpnService: Not a numeric address: 94web.com
08-18 05:25:35.742 21590 21590 E org.amnezia.vpn: Not starting debugger since process cannot load the jdwp agent.
08-18 05:25:35.974 21590 21611 E OpenGLRenderer: Unable to match the desired swap behavior.`
